tests/flytekit/unit/core/test_type_hints.py

```diff
@@ -329,6 +329,11 @@ def my_wf(a: typing.List[int]) -> (int, str):
     x = my_wf(a=[5, 6])
     assert x == (15, "world-7world-8")
 
+    a = [5, 6]
+    x, y = maptask(t1, metadata=TaskMetadata(retries=1))(a=a)
```

Comment:
what error do we get in the following cases
- if we try to bind a non-list type so i.e, the input `a` is not a list but an `int` 
- If we try to bind a str to a (not a list)
- if we bind a list of str to `a`

Reply:
updated with additional test cases